torch/csrc/jit/script/python_sugared_value.h

```diff
@@ -193,18 +193,128 @@ struct VISIBILITY_HIDDEN OverloadedFunctionValue : public SugaredValue {
   std::vector<StrongFunctionPtr> compiled_overloads_;
 };
 
-// defines how modules/methods behave inside the script subset.
-// for now this does not have any interaction with python.
-// in the future, we will add the ability to resolve `self.foo` to python
-// {functions, modules, contants} so this SugaredValue is defined here
-// anticipating we will eventually need to replace Module with a py::object
-// holding the actual nn.Module class.
+// This holds all relevant Python state that ModuleValue needs. I
+struct VISIBILITY_HIDDEN ModuleMetadata {
```

Comment:
/* Trying to grasp what's going on in this change */ Why is this in `python_sugared_value.h`? It looks to me that it's an independent from Python concept? Do I misunderstand something?